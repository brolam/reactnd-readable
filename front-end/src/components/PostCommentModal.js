import React from 'react'
import PropTypes from 'prop-types'

function PostCommentModal(props) {
  const { post, comment } = props
  return (
    <div id="commentModal" className="modal modal-open">
      <div className="modal-dialog">
        <div className="modal-heard">
          <span onClick={props.onClickBackButton} />
          <h1>
            {isNewComment(comment) ? 'New comment' : 'Edit comment'}
          </h1>
        </div>
        <div className="modal-content modal-comment">
          <textarea id="commentModalBody" placeholder="Body comment" />
          <div className="modal-footer">
            <button className={"save-button " + post.category} href="/"
              onClick={e => { }}>Save</button>
          </div>
        </div>
      </div>
    </div>
  )
}

function isNewComment(comment) {
  return comment.id ? true : false
}

PostCommentModal.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
  }),
  comment: PropTypes.shape({}).isRequired
}

export default PostCommentModal