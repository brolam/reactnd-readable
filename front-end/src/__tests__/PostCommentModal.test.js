import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PostCommentModal from '../components/PostCommentModal'

test('render without error', () => {
  ReactDOM.render(<PostCommentModal post={postUdacity} comment={comment} />, document.createElement('div'));
})

test('last Snapshot to new comment', () => {
  const postCommentModal = renderer.create(
    <PostCommentModal
      post={postUdacity}
      comment={{}}
    />);
  let tree = postCommentModal.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot to edit comment', () => {
  const postCommentModal = renderer.create(
    <PostCommentModal
      post={postUdacity}
      comment={comment}
    />);
  let tree = postCommentModal.toJSON();
  expect(tree).toMatchSnapshot();
})

const comment = {
  id: '894tuq4ut84ut8v4t8wun89g',
  parentId: "7ni6ok3ym7mf1p33lnez",
  timestamp: 1468166872634,
  body: 'Hi there! I am a COMMENT.',
  author: 'thingtwo',
  voteScore: 6,
  deleted: false,
  parentDeleted: false
}

const postUdacity = {
  category: "udacity",
  id: "7ni6ok3ym7mf1p33lnez",
  title: "Udacity is the best place to learn technology.",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}
