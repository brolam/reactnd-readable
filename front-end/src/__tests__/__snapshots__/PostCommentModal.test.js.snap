// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot to edit comment 1`] = `
<div
  className="modal modal-open"
  id="commentModal"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-heard"
    >
      <span
        onClick={undefined}
      />
      <h1>
        New comment
      </h1>
    </div>
    <div
      className="modal-content modal-comment"
    >
      <textarea
        id="commentModalBody"
        placeholder="Body comment"
      />
      <div
        className="modal-footer"
      >
        <button
          className="save-button udacity"
          href="/"
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`last Snapshot to new comment 1`] = `
<div
  className="modal modal-open"
  id="commentModal"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-heard"
    >
      <span
        onClick={undefined}
      />
      <h1>
        Edit comment
      </h1>
    </div>
    <div
      className="modal-content modal-comment"
    >
      <textarea
        id="commentModalBody"
        placeholder="Body comment"
      />
      <div
        className="modal-footer"
      >
        <button
          className="save-button udacity"
          href="/"
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
